PAPP 21767 - add FIPS compliance

shiyingtu opened the pull request:
- Mark as FIPS Compliant
- Fix linting tests
- [Fix C901 'MolochConnector._handle_get_pcap' is too complex (30)](https://github.com/splunk-soar-connectors/moloch/pull/1/commits/d50bb3c395c30d9ec53222701ddee4edcfffeef5)

Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

## Pull Request Checklist

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
- [ ] Limit your pull request to one application and submit multiple pull requests if necessary.
## Pull Request Type

#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.


## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

brian-phantom committed:
Merge branch 'next' into 'master'

Moloch : Next to Master

See merge request phantom-apps/phmoloch!10

ricardo-phantom committed:
Bumped up the version of phmoloch from 2.0.1 to 2.0.2

shiyingtu committed:
Merge branch 'next' into PAPP-21767-add-fips-compliance

shiyingtu committed:
Fix linting tests

shiyingtu committed:
Mark Moloch as FIPS Compliant in app metadata

shiyingtu committed:
Fix C901 'MolochConnector._handle_get_pcap' is too complex (30)

- Issue: this warning shows up because the function creates a lot
of if/else branches, making the function more complex
- Fix: Avoid if/else as possible, use for loop and join to finish
the task

shiyingtu commented:
Static failing tests:

- Min Platform Version

tonyc-phantom commented on moloch_connector.py:
```diff
@@ -245,7 +245,7 @@ def _make_rest_call(self, endpoint, action_result, headers=None, params=None, da
         try:
             url = '{url}{endpoint}'.format(url=self._server_url, endpoint=endpoint)
-        except Exception as e:
+        except:
```
Don't we want to have "except Exception" here?

shiyingtu replied:
Hi Tony - Exception e is not used anywhere, which is detected by linting tests.

tonyc-phantom replied:
Oh, what I meant was not using a bare exception. Instead of "except Exception as e", just "except Exception". I only mentioned it because @rgil-splunk mentioned it to me before in one of his reviews of my code. I believe the bare except would catch certain exceptions we wouldn't want to catch such as KeyboardInterrupt and SystemExit. I don't know if this would ultimately cause an issue though, perhaps Ronald can comment?

rgil-splunk replied:
what you mentioned Tony pretty much covers it :)

while there is a low probability of it causing an issue, it would be difficult to debug and pinpoint what happened if there is an issue given it would happen during failure / exit states

link for reference: https://stackoverflow.com/questions/54948548/what-is-wrong-with-using-a-bare-except/54948581#54948581

shiyingtu replied:
Make sense! Thanks Ronald and Tony. Let me fix this

shiyingtu committed:
Avoid to use bare except

- Issue: bare except would catch all the exceptions including
Ctrl-c (Keyboard Interrupt) which would be hard to debug
- Fix: add specific Exception
